test/cypress/integration/widgets/deployment_button_spec.ts

```diff
@@ -226,6 +228,32 @@ describe('Create Deployment Button widget', () => {
             );
             cy.get('@string_constraint_pattern').should('have.class', 'error');
         });
+
+        it('should open Missing Secrets Error message when deploying blueprint with missing secrets', () => {
+            const secretName = 'test';
+
+            selectBlueprintInModal('required_secrets');
+            cy.stageRequest(`/console/sp/secrets/${secretName}`, 'DELETE', { failOnStatusCode: false });
+
+            cy.get('input[name=deploymentName]').type('blahBlahBlah');
+            cy.contains('.modal button', 'Deploy').click();
+            cy.get('form.error .error').within(() => {
+                cy.get('.header').should('have.text', 'Missing Secrets Error');
+                cy.get('p').should('have.text', 'The following required secrets are missing in this tenant:');
+                cy.get('.item').should('have.text', secretName);
+            });
+
+            cy.contains('form.error button', 'Add Missing Secrets').click();
+            cy.contains('.secretsModal button', 'Add').click();
+            cy.get('.secretsModal .error .header').should('have.text', 'Errors in the form');
+            cy.get('.secretsModal .error li').should('have.text', 'Please provide values for secrets');
+
+            cy.get('.secretsModal input').type('aaa');
+            cy.interceptSp('PUT', `/secrets/${secretName}`, { statusCode: 200 }).as('addSecrets');
```

Comment:
The 3rd argument will work like a static response, so Cypress will intercept this request and always respond with statusCode = 200.

See `interceptSp` definition + see `cy.intercept` documentation (https://docs.cypress.io/api/commands/intercept#Syntax)

I think you should have here:
```suggestion
            cy.interceptSp('PUT', `/secrets/${secretName}`).as('addSecrets');
```
That way Cypress will just listen for `PUT /secrets/${secretName}` requests and won't change anything inside.

Reply:
I fixed in the following PR: https://github.com/cloudify-cosmo/cloudify-stage/pull/1705